server/src/test/java/org/apache/druid/server/http/DataSourcesResourceTest.java

```diff
@@ -621,7 +622,8 @@ public void testMarkAsUnusedAllSegmentsInDataSource()
       Assert.assertNotNull(response.getEntity());
       Assert.assertTrue(response.getEntity().toString().contains("java.lang.IllegalArgumentException"));
     }
-    catch (IllegalArgumentException ignore) {
+    catch (DruidException ignore) {
+      Assert.assertEquals(DruidException.Category.INVALID_INPUT, ignore.getCategory());
       // expected
     }
```

Comment:
Can you switch this around to `DruidExceptionMatcher.invalidInput().assertThrowsAndMatches()`